internal/hmyapi/blockchain.go

```diff
@@ -86,9 +87,10 @@ func (s *PublicBlockChainAPI) GetStorageAt(ctx context.Context, address common.A
 // GetBalance returns the amount of Nano for the given address in the state of the
 // given block number. The rpc.LatestBlockNumber and rpc.PendingBlockNumber meta
 // block numbers are also allowed.
-func (s *PublicBlockChainAPI) GetBalance(ctx context.Context, address common.Address, blockNr rpc.BlockNumber) (*hexutil.Big, error) {
+func (s *PublicBlockChainAPI) GetBalance(ctx context.Context, address string, blockNr rpc.BlockNumber) (*hexutil.Big, error) {
 	// TODO: currently only get latest balance. Will add complete logic later.
-	return s.b.GetBalance(address)
+	adr := internal_common.ParseAddr(address)
```

Comment:
What about malformed adresses?

Reply:
i was thinking about that but i looked at deeper logic it can always return any address.

If the address input is specified as malformed it just returns 0x0000000000000000000000000000000000000000

Reply:
fmt.Println(ParseAddr("minh3").Hex()) returns "0x0000000000000000000000000000000000000000"

Reply:
That address maybe used for burning tokens, so it may also have balances, please add a todo to better handle this